mesonbuild/modules/java.py

```diff
@@ -54,6 +56,18 @@ def __get_java_compiler(self, state: ModuleState) -> Compiler:
         KwargInfo('package', str, default=None))
     def generate_native_headers(self, state: ModuleState, args: T.Tuple[T.List[mesonlib.FileOrString]],
                                 kwargs: T.Dict[str, T.Optional[str]]) -> ModuleReturnValue:
+        return self.native_headers(state, args, kwargs)
```

Comment:
Duplicating the type checking is unfortunate. Also, this will print the FeatureNew for `native_headers` even if you use `generate_native_headers` with a minimum meson of `>=0.63.0`.

Reply:
I see. Let me move the code into a shared function then.